### PR TITLE
Fix netskope issue

### PR DESCRIPTION
**Pull request by RosenbergYehuda**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes:https://jira-dc.paloaltonetworks.com/browse/XSUP-74841
## Description
Fixed an issue where audit events were not ingested.
Added the *start_time* and *end_time* arguments to the ***netskope-get-events*** command, allowing events to be fetched for a specific time window.

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by marketplace-ai-reviewer** [comment]
🤖 Analysis started. Please wait for results...

**Comment by marketplace-ai-reviewer** [comment]
<details>
<summary><strong>🤖 AI Review Disclaimer</strong></summary>

This review was generated by an AI-powered tool and may contain inaccuracies. Please be advised, and we extend our sincere apologies for any inconvenience this may cause.
</details>

**Review by marketplace-ai-reviewer — COMMENTED** [review]
Thanks for submitting your PR!
Before moving on with the detailed review, please take a moment to address the following general comments.

Once you've improved your PR according to the comments below, re-trigger the review by adding the `ready-for-ai-review` label or commenting `@marketplace-ai-reviewer start review`.


- The PR description mentions fixing an issue with audit events ingestion, but the code also adds new `start_time` and `end_time` arguments to the `netskope-get-events` command. Please update the PR title and description to accurately reflect this new functionality, or split the PR if the changes are unrelated.

**Comment by RosenbergYehuda** [comment]
@marketplace-ai-reviewer re-review

**Comment by RosenbergYehuda** [comment]
@marketplace-ai-reviewer start review

**Comment by marketplace-ai-reviewer** [comment]
🤖 Analysis started. Please wait for results...

File: Packs/Netskope/Integrations/NetskopeEventCollector_v2/NetskopeEventCollector_v2.py [hunk 1]
```diff
@@ -486,8 +495,48 @@ async def _send_page_to_xsiam(events):
...
+        results: list = []
+        offset = init_offset
+        while offset < max_offset:
+            # Cap total at `limit` so the caller's `events_count == limit` check checkpoints next_fetch_offset.
+            page_size = min(request_limit, max_offset - offset)
+            try:
+                page_result = await _handle_page(request_params | {"offset": offset, "limit": page_size})
+            except Exception as e:
+                # Record the failure (don't abort the window) so the caller checkpoints and retries this offset.
+                demisto.error(f"[Fetch] type={type}: sequential page failed at {offset=}: {str(e)}")
```
Comment (marketplace-ai-reviewer) [inline Packs/Netskope/Integrations/NetskopeEventCollector_v2/NetskopeEventCollector_v2.py:525]: Every `except` block that logs an error MUST include the full traceback. Please add `demisto.debug(traceback.format_exc())` here.

```python
                demisto.error(f"[Fetch] type={type}: sequential page failed at {offset=}: {str(e)}")
                demisto.debug(traceback.format_exc())
```

File: Packs/Netskope/Integrations/NetskopeEventCollector_v2/NetskopeEventCollector_v2.py [hunk 2]
```diff
@@ -629,8 +678,36 @@ async def handle_fetch_and_send_all_events(
 async def get_events_command_async(
     client: Client, args: dict[str, Any], last_run: dict, should_push_events: bool = False
 ) -> CommandResults:
-    """Manual netskope-get-events command: fetch a small batch, optionally push it, and display it."""
+    """Manual netskope-get-events command: fetch a small batch, optionally push it, and display it.
```
Comment (marketplace-ai-reviewer) [inline Packs/Netskope/Integrations/NetskopeEventCollector_v2/NetskopeEventCollector_v2.py:682]: Since this integration fetches multiple event types, the `netskope-get-events` command MUST accept an `event_type` argument so the operator can scope the manual fetch to a single type during debugging. Please add this argument to the command in the YAML and handle it here. (See §6.2.2)

File: Packs/Netskope/Integrations/NetskopeEventCollector_v2/NetskopeEventCollector_v2.py [hunk 3]
```diff
@@ -629,8 +678,36 @@ async def handle_fetch_and_send_all_events(
...
+    # 1 day ago, end to now). For audit this filters by insertion time (insertionstarttime/endtime).
+    if args.get("start_time") or args.get("end_time"):
+        # `arg_name` makes arg_to_datetime raise a clear ValueError (surfaced to the user) on
+        # unparseable input, so the values below are always valid datetimes.
+        start_arg = arg_to_datetime(args.get("start_time") or "1 day", arg_name="start_time")
+        end_arg = arg_to_datetime(args.get("end_time") or "now", arg_name="end_time")
+        start_epoch = str(int(start_arg.timestamp()))  # type: ignore[union-attr]
+        end_epoch = str(int(end_arg.timestamp()))  # type: ignore[union-attr]
+        # Validate the time window
+        if args.get("start_time") and args.get("end_time") and int(end_epoch) <= int(start_epoch):
```
Comment (marketplace-ai-reviewer) [inline Packs/Netskope/Integrations/NetskopeEventCollector_v2/NetskopeEventCollector_v2.py]: Simplify time window validation to catch invalid defaults.

File: Packs/Netskope/Integrations/NetskopeEventCollector_v2/NetskopeEventCollector_v2_test.py [hunk 4]
```diff
@@ -478,6 +610,82 @@ async def test_get_events_count(mocker):
...
+        send_to_xsiam=False,
+    )
+
+    assert not failures
+    # Exactly two API calls: the full page, then the short page (which stops the loop).
+    assert data_mock.call_count == 2
+    fetched = [ev for page in success for ev in page]
+    assert {e["_id"] for e in fetched} == {"1", "2", "3"}
+
+
```
Comment (marketplace-ai-reviewer) [inline Packs/Netskope/Integrations/NetskopeEventCollector_v2/NetskopeEventCollector_v2_test.py:688]: Consider adding a test case for the exact `limit` boundary condition.

File: Packs/Netskope/Integrations/NetskopeEventCollector_v2/NetskopeEventCollector_v2.yml [hunk 5]
```diff
@@ -74,6 +74,10 @@ script:
     - description: The maximum number of alerts to return.
       name: limit
       defaultValue: 10
+    - description: 'Start of the time window to fetch events for. Accepts a date (e.g. "2026-08-11T00:00:00Z"), a relative time (e.g. "3 days"), or a Unix epoch timestamp (e.g. 1786513755). If omitted, the command uses the instance''s last-run position (default behavior).'
```
Comment (marketplace-ai-reviewer) [inline Packs/Netskope/Integrations/NetskopeEventCollector_v2/NetskopeEventCollector_v2.yml]: Argument descriptions must start with "The " and end with a period.

```suggestion
    - description: 'The start of the time window to fetch events for. Accepts a date (e.g. "2026-08-11T00:00:00Z"), a relative time (e.g. "3 days"), or a Unix epoch timestamp (e.g. 1786513755). If omitted, the command uses the instance''s last-run position (default behavior).'
```

File: Packs/Netskope/Integrations/NetskopeEventCollector_v2/NetskopeEventCollector_v2.yml [hunk 6]
```diff
@@ -74,6 +74,10 @@ script:
...
+      name: start_time
```
Comment (marketplace-ai-reviewer) [inline Packs/Netskope/Integrations/NetskopeEventCollector_v2/NetskopeEventCollector_v2.yml:84]: Since this argument is parsed as a date in the code using `arg_to_datetime`, its type should be explicitly set to `date` (type 3).

```suggestion
      name: start_time
      type: 3
```

File: Packs/Netskope/Integrations/NetskopeEventCollector_v2/NetskopeEventCollector_v2.yml [hunk 7]
```diff
@@ -74,6 +74,10 @@ script:
...
+    - description: 'End of the time window to fetch events for. Accepts a date (e.g. "2026-08-12T00:00:00Z"), a relative time (e.g. "1 day"), or a Unix epoch timestamp (e.g. 1786600155). Defaults to now.'
```
Comment (marketplace-ai-reviewer) [inline Packs/Netskope/Integrations/NetskopeEventCollector_v2/NetskopeEventCollector_v2.yml]: Argument descriptions must start with "The " and end with a period. Additionally, default values must live in the `defaultValue` field, not in the description.

```suggestion
    - description: 'The end of the time window to fetch events for. Accepts a date (e.g. "2026-08-12T00:00:00Z"), a relative time (e.g. "1 day"), or a Unix epoch timestamp (e.g. 1786600155).'
```

File: Packs/Netskope/Integrations/NetskopeEventCollector_v2/NetskopeEventCollector_v2.yml [hunk 8]
```diff
@@ -74,6 +74,10 @@ script:
...
+      name: end_time
```
Comment (marketplace-ai-reviewer) [inline Packs/Netskope/Integrations/NetskopeEventCollector_v2/NetskopeEventCollector_v2.yml:86]: Since this argument is parsed as a date in the code using `arg_to_datetime`, its type should be explicitly set to `date` (type 3). Also, since the description mentions a default value of "now", it should be defined here.

```suggestion
      name: end_time
      type: 3
      defaultValue: now
```

File: Packs/Netskope/Integrations/NetskopeEventCollector_v2/NetskopeEventCollector_v2.yml [hunk 9]
```diff
@@ -74,6 +74,10 @@ script:
...
     description: Returns events extracted from SaaS traffic and or logs.
```
Comment (marketplace-ai-reviewer) [inline Packs/Netskope/Integrations/NetskopeEventCollector_v2/NetskopeEventCollector_v2.yml]: According to XSOAR guidelines, the `get-events` command description must end with a specific mandatory warning sentence regarding development and debugging use.

```suggestion
      description: Returns events extracted from SaaS traffic and or logs. Use this command for development and debugging only, as it may produce duplicate events, exceed API rate limits, or disrupt the fetch mechanism.
```

File: Packs/Netskope/Integrations/NetskopeEventCollector_v2/README.md [hunk 10]
```diff
@@ -34,6 +34,8 @@ Returns events extracted from SaaS traffic and or logs.
 | --- | --- | --- |
 | limit | The maximum number of alerts to return (default: 10). | Optional |
 | should_push_events | Set this argument to True in order to create events, otherwise the command will only display them. | Optional |
+| start_time | Start of the time window to fetch events for. Accepts a date (e.g. "2026-08-11T00:00:00Z"), a relative time (e.g. "3 days"), or a Unix epoch timestamp (e.g. 1786513755). If omitted, the command uses the instance's last-run position (default behavior). | Optional |
```
Comment (marketplace-ai-reviewer) [inline Packs/Netskope/Integrations/NetskopeEventCollector_v2/README.md]: It appears the `#### Command example` section was not updated to demonstrate the usage of these new arguments.

Please add an example using the new arguments (e.g., `!netskope-get-events start_time="3 days ago"`) to the `command_examples.txt` file and regenerate the README using the SDK. This ensures the documentation remains consistent and auto-generated. For more details, see [Command Examples - Integration Documentation](https://xsoar.pan.dev/docs/documentation/readme_file#command-examples---integration-documentation).

```bash
demisto-sdk generate-docs --insecure -e Packs/Netskope/Integrations/NetskopeEventCollector_v2/command_examples.txt -i Packs/Netskope/Integrations/NetskopeEventCollector_v2/NetskopeEventCollector_v2.yml
```

**Comment by marketplace-ai-reviewer** [comment]
<details>
<summary><strong>🤖 AI Review Disclaimer</strong></summary>

This review was generated by an AI-powered tool and may contain inaccuracies. Please be advised, and we extend our sincere apologies for any inconvenience this may cause.
</details>

**Review by marketplace-ai-reviewer — COMMENTED** [review]
Hi! Thanks for your contribution to the Netskope pack. I've reviewed your changes and have a few suggestions to help align this with XSOAR standards. The main items include bumping the pack version to minor (since new arguments were added), adding an `event_type` argument to `get-events` for easier debugging, and ensuring exceptions log the full traceback. You'll also need to tweak some YAML argument types, descriptions, and regenerate the README examples. Thanks for the great work!

Additionally, please address the following file-level notes:
- `Packs/Netskope/pack_metadata.json`: The vendor name must appear in the `keywords` list.
- `Packs/Netskope/ReleaseNotes/4_2_8.md`: The release notes describe adding new arguments (`start_time` and `end_time`), which is a feature addition. However, the version bump from the filename (`4_2_8.md`) indicates a patch bump (revision). Feature additions require a **minor** version bump (e.g., `4.3.0`).

Please run `demisto-sdk update-release-notes -i Packs/Netskope -u minor` to regenerate the version and release-notes filename consistently. Refer to the [General](https://xsoar.pan.dev/docs/documentation/release-notes#general) release notes documentation for more details.

 @marketplace-ai-reviewer, @JasBeilin, @RosenbergYehuda please review and approve the results generated by the AI Reviewer by responding 👍 on this comment.

File: Packs/Netskope/Integrations/NetskopeEventCollector_v2/NetskopeEventCollector_v2.py [hunk 11]
```diff
@@ -486,8 +495,49 @@ async def _send_page_to_xsiam(events):
             # get-events path: caller needs the actual events
             return events
 
+    def _page_result_len(page_result) -> int:
```
Comment (DanielTal87) [inline Packs/Netskope/Integrations/NetskopeEventCollector_v2/NetskopeEventCollector_v2.py]: I would add type to the `page_result` since you use it in the return 


```suggestion
        def _page_result_len(page_result: int | list[dict]) -> int:
```

File: Packs/Netskope/Integrations/NetskopeEventCollector_v2/NetskopeEventCollector_v2.py [hunk 12]
```diff
@@ -629,8 +679,35 @@ async def handle_fetch_and_send_all_events(
...
+    # start_time the instance's real last_run is used (default behavior). For audit this filters by
+    # insertion time (insertionstarttime/endtime). `arg_name` makes arg_to_datetime raise a clear
+    # error on unparseable input, so the parsed values are always valid datetimes.
+    if args.get("start_time"):
+        start_arg = arg_to_datetime(args.get("start_time"), arg_name="start_time")
+        end_arg = arg_to_datetime(args.get("end_time") or "now", arg_name="end_time")
+        start_epoch = str(int(start_arg.timestamp()))  # type: ignore[union-attr]
+        end_epoch = str(int(end_arg.timestamp()))  # type: ignore[union-attr]
+        if int(end_epoch) <= int(start_epoch):
+            return_error(f"'end_time' ({end_epoch}) must be after 'start_time' ({start_epoch}).")
```
Comment (DanielTal87) [inline Packs/Netskope/Integrations/NetskopeEventCollector_v2/NetskopeEventCollector_v2.py]: you convert / cast int to str and then in the if cast again to int 
you can just keep it int 

```suggestion
        start_ts = int(start_arg.timestamp())  # type: ignore[union-attr]
        end_ts = int(end_arg.timestamp())  # type: ignore[union-attr]
        if end_ts <= start_ts:
            return_error(f"'end_time' ({end_ts}) must be after 'start_time' ({start_ts}).")
        start_epoch = str(start_ts)
        end_epoch = str(end_ts)
```

**Review by DanielTal87 — APPROVED** [review]
